src/Components/Endpoints/src/Builder/ResourceCollectionUrlEndpoint.cs

```diff
@@ -118,6 +118,35 @@ private static string ComputeFingerprintSuffix(ResourceAssetCollection resourceC
         {
             var url = resource.Url;
             AppendToHash(incrementalHash, buffer, ref rented, url);
+
+            // For non-fingerprinted assets (those without a 'label' property), we need to include
+            // the integrity hash in the fingerprint calculation. This ensures that if the content
+            // changes between builds, the resource-collection fingerprint also changes.
+            if (resource.Properties != null)
+            {
+                var hasLabel = false;
+                string? integrity = null;
+                foreach (var property in resource.Properties)
+                {
+                    if (property.Name.Equals("label", StringComparison.OrdinalIgnoreCase))
+                    {
+                        hasLabel = true;
+                        // No need to continue if we found a label - we won't use integrity
+                        break;
+                    }
+                    else if (property.Name.Equals("integrity", StringComparison.OrdinalIgnoreCase))
+                    {
+                        integrity = property.Value;
+                        // Continue searching in case there's also a label property
+                    }
+                }
```

Comment:
Move this into a helper method that returns a string and then have the check be against the returned string. If there is more data we need to hash we will return a non-null string. Otherwise, we return null and skip.